### PR TITLE
Move BlockManager to node/blockstorage

### PR DESCRIPTION
**Pull request by maflcko**
`BlockManager` is responsible for reading and writing block(headers). So move it to the existing `blockstorage` module in `node`. Also, move validation code unrelated to block-storage out from `BlockManager`.

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -718,7 +718,9 @@ bool MemPoolAccept::PreChecks(ATMPArgs& args, Workspace& ws)
     if (!CheckSequenceLocks(m_active_chainstate.m_chain.Tip(), m_view, tx, STANDARD_LOCKTIME_VERIFY_FLAGS, &lp))
         return state.Invalid(TxValidationResult::TX_PREMATURE_SPEND, "non-BIP68-final");
 
-    if (!Consensus::CheckTxInputs(tx, state, m_view, m_active_chainstate.m_blockman.GetSpendHeight(m_view), ws.m_base_fees)) {
+    assert(m_active_chainstate.m_blockman.LookupBlockIndex(m_view.GetBestBlock()) == m_active_chainstate.m_chain.Tip());
```
Comment (promag) [inline src/validation.cpp:715]: fa646a5ae9b39b7987d5701aec55a54ae4ec80ed

nit, move after L706.

File: src/validation.h [hunk 2]
```diff
@@ -427,6 +428,7 @@ class BlockManager
      *                                     which we've downloaded all transactions.
      */
     bool LoadBlockIndex(
+        CBlockIndex*& best_invalid,
```
Comment (promag) [inline src/validation.h]: faabaa8654c9446846e79ac5ae2edf70e0ae6d98

Update doc above? Make it the 2nd arg?

**Review by promag — COMMENTED** [review]
Concept ACK.

File: src/validation.cpp (hunk 1 above)
Comment (MarcoFalke) [reply src/validation.cpp]: thx, done

File: src/validation.h (hunk 2 above)
Comment (MarcoFalke) [reply src/validation.h]: thx, done

**Comment by MarcoFalke** [comment]
I've split out the first commit to https://github.com/bitcoin/bitcoin/pull/23630, since it is not really related to moving code around.

Please review that before this pull.

**Comment by MarcoFalke** [comment]
Rebased

**Comment by MarcoFalke** [comment]
Rebased

**Review by ryanofsky — APPROVED** [review]
Code review obvious ACK fa7efc915b87ec56ca1cc0bad7d8f79591bfa099

**Pull request merged**